packages/grid/x-data-grid-premium/src/hooks/features/aggregation/wrapColumnWithAggregation.tsx

```diff
@@ -0,0 +1,290 @@
+  // Then we don't wrap the column value formatter
+  if (!aggregationRule.aggregationFunction.valueFormatter) {
+    return valueFormatter;
+  }
+
+  const wrappedValueFormatter: GridColDef['valueFormatter'] = (params) => {
+    if (params.id != null) {
+      const cellAggregationResult = getCellAggregationResult(params.id, params.field);
+      if (cellAggregationResult != null) {
+        return aggregationRule.aggregationFunction.valueFormatter!(params);
+      }
+    }
+
+    if (valueFormatter) {
+      return valueFormatter(params);
+    }
+
+    return params.value;
+  };
+
+  return wrappedValueFormatter;
+};
+
+const getAggregationValueWrappedRenderCell: ColumnPropertyWrapper<'renderCell'> = ({
+  value: renderCell,
+  aggregationRule,
+  getCellAggregationResult,
+}) => {
+  const wrappedRenderCell: GridColDef['renderCell'] = (params) => {
+    const cellAggregationResult = getCellAggregationResult(params.id, params.field);
+    if (cellAggregationResult != null) {
+      if (!renderCell) {
+        if (cellAggregationResult.position === 'footer') {
+          return <GridFooterCell {...params} />;
+        }
+
+        return params.formattedValue;
+      }
+
+      const aggregationMeta: GridAggregationCellMeta = {
+        hasCellUnit: aggregationRule!.aggregationFunction.hasCellUnit ?? true,
+        aggregationFunctionName: aggregationRule!.aggregationFunctionName,
+      };
+
+      return renderCell({ ...params, aggregation: aggregationMeta });
+    }
+
+    if (!renderCell) {
+      return params.formattedValue;
+    }
+
+    return renderCell(params);
+  };
+
+  return wrappedRenderCell;
+};
+
+/**
+ * Skips the filtering for aggregated rows
+ */
+const getWrappedFilterOperators: ColumnPropertyWrapper<'filterOperators'> = ({
+  value: filterOperators,
+  getCellAggregationResult,
+}) =>
+  filterOperators!.map((operator) => {
+    return {
+      ...operator,
+      getApplyFilterFn: (filterItem, column) => {
+        const originalFn = operator.getApplyFilterFn(filterItem, column);
+        if (!originalFn) {
+          return null;
+        }
+
+        return (params) => {
+          if (getCellAggregationResult(params.id, params.field) != null) {
+            return true;
+          }
+
+          return originalFn(params);
+        };
+      },
+    };
+  });
+
+/**
+ * Add the aggregation method around the header name
+ */
+const getWrappedRenderHeader: ColumnPropertyWrapper<'renderHeader'> = ({
+  apiRef,
+  value: renderHeader,
+  aggregationRule,
+}) => {
+  const wrappedRenderCell: GridColDef['renderHeader'] = (params) => {
+    const baseHeaderLabel = params.colDef.headerName ?? params.colDef.field;
+    const aggregationLabel = getAggregationFunctionLabel({ apiRef, aggregationRule });
+
+    if (!renderHeader) {
+      return (
+        <GridColumnHeaderTitle
+          label={`${aggregationLabel}(${baseHeaderLabel})`}
```

Comment:
Wouldn't be easy to wrap the `headerName` property? You're duplicating code from `GridColumnHeaderItem`.

https://github.com/mui/mui-x/blob/22b7dd40455066dda5d8b41242a19d2a65090c73/packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx#L252-L253

Reply:
No, see https://github.com/mui/mui-x/pull/4208#discussion_r889999299
We reuse the header name for grouping and it's very weird if it also wraps with the aggregation there